src/tests/system/tests/test_idp.py

```diff
@@ -0,0 +1,147 @@
+"""
+SSSD IdP provider Test Cases
+
+:requirement: IDP
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.keycloak import Keycloak
+from sssd_test_framework.topology import KnownTopology
+
+configurations = {
+    "enabled": "true",
+    "idp_type": "keycloak:https://master.keycloak.test:8443/auth/admin/realms/master/",
+    "id_provider": "idp",
+    "auto_private_groups": "true",
+    "use_fully_qualified_names": "true",
+    "debug_level": "9",
+    "idp_client_id": "myclient",
+    "idp_client_secret": "ClientSecret123",
+    "idp_token_endpoint": "https://master.keycloak.test:8443/auth/realms/master/protocol/openid-connect/token",
+    "idp_userinfo_endpoint": "https://master.keycloak.test:8443/auth/realms/master/protocol/openid-connect/userinfo",
+    "idp_device_auth_endpoint": "https://master.keycloak.test:8443/auth/realms/master/protocol/openid-connect/auth/device",
+    "idp_id_scope": "profile",
+    "idp_auth_scope": "openid profile email",
+}
```

Comment:
We can include this in the framework topology. See how it is done e.g. for IPA.

You need to include domains in the topology definition: https://github.com/SSSD/sssd-test-framework/blob/master/sssd_test_framework/topology.py#L71

Then set the configuration in the host object (or mhc.yaml): https://github.com/SSSD/sssd-test-framework/blob/master/sssd_test_framework/hosts/ipa.py#L59-L63

Reply:
See https://github.com/SSSD/sssd-test-framework/pull/187.